enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,129 @@
+2. **As a cluster administrator on a baremetal platform**, I want a reliable solution to handle HTTP01 challenges for the API endpoint, even when the endpoint is not managed by OpenShift Ingress, so that I can avoid manual workarounds.
+3. **As a developer**, I want a simple deployment mechanism for the HTTP01 challenge proxy, so that I can easily integrate it into my existing cluster setup.
+
+### Proposal
+
+The HTTP01 Challenge Proxy will be implemented via DaemonSet running on the cluster. It will:
+
+- Redirect HTTP traffic from the API endpoint (`api.cluster.example.com`) on port 80 to the OpenShift Ingress Routers.
+- Use `nftables` for traffic redirection from `API:80` to `PROXY:8888`.
+- Be deployed using a manifest that includes all necessary configurations.
+
+The proxy will ensure compatibility with various OCP topologies, including SNO, MNO, and Compact clusters, addressing the challenges of HTTP01 validation for the API endpoint.
+
+### API Extensions
+
+A new CR type may be created and can be applied to clusters.  This new typed will be stored in the [openshift/api](https://github.com/openshift/api) repo.
+
+### Implementation Details/Notes/Constraints
+
+- The proxy will be deployed as a DaemonSet to ensure it runs on all nodes which may host the API VIP in the cluster.
+- The nftables rules will be added to the nodes. The proxy will listen on port 8888 and redirect traffic to the OpenShift Ingress Routers.
+- The implementation relies on `nftables` for traffic redirection, which must be supported and enabled on the cluster nodes.
+- The demo deployment manifest for the proxy is available [here](https://github.com/mvazquezc/cert-mgr-http01-proxy/blob/main/manifests/deploy-in-ocp.yaml).
+- An example implementation can be found in this [repository](https://github.com/mvazquezc/cert-mgr-http01-proxy/tree/main).
+
+### Design Details
+
+- **Proxy Deployment**: The proxy will be deployed using a Kubernetes DaemonSet. The daemonset will implement an nftable rule via pod that runs to completion.
+- **Traffic Redirection**: This will use `nftables` rules to redirect incoming traffic on `API:80` to `PROXY:8888`.
+- **Security**: The proxy will only handle HTTP traffic for the HTTP01 challenge and will not interfere with other traffic or services.
+- **Monitoring**: Logs and metrics will be exposed to help administrators monitor the proxy's behavior and troubleshoot issues.
+
+### Drawbacks
+
+1. **Dependency on nftables**: The solution relies on `nftables`, which may not be available or enabled on all environments.
+2. **Additional Resource Usage**: Running the proxy as a DaemonSet introduces additional resource usage on the cluster nodes while the proxy pod is applying its nftable rules.
+3. **Complexity**: The solution adds another component to the cluster, which may increase operational complexity.
+
+### Alternatives
+
+None
```

Comment:
There are _no_ alternatives to this approach that you've considered and ruled out?

Reply:
I didn't list them here but @mvazquezc put together some alternative proposals in this document (linked in the Jira): https://docs.google.com/presentation/d/1mJ1pnsPiEwb-U5lHwhM2UkyRmkkLeYxj3cfE4F7dOx0/edit#slide=id.g547716335e_0_260

Should I list these other proposals that aren't the PoC's implementation here?

Reply:
My understanding is that EPs are generally used as the source of truth when making changes like this across OCP and are linked to when creating feature gates.

I would at least include a high-level summary of each of the alternative approaches and why they were insufficient.